lumicks/pylake/tests/test_imaging_confocal/test_scan.py

```diff
@@ -247,23 +247,65 @@ def test_deprecated_plotting(test_scans):
         scan.plot_rgb()
 
 
-def test_export_tiff(tmpdir_factory, test_scans):
+@pytest.mark.parametrize(
+    "scanname, tiffname",
+    [
+        ("fast Y slow X", "single_frame.tiff"),
+        ("fast Y slow X multiframe", "multi_frame.tiff"),
+    ],
+)
+def test_export_tiff(scanname, tiffname, tmp_path, test_scans):
     from os import stat
 
-    tmpdir = tmpdir_factory.mktemp("pylake")
+    def grab_tags(file):
+        import tifffile
+        from ast import literal_eval
+
+        tiff_tags = []
+        with tifffile.TiffFile(file) as tif:
+            for page in tif.pages:
+                page_tags = {}
+                for tag in page.tags.values():
+                    name, value = tag.name, tag.value
+                    try:
+                        page_tags[name] = literal_eval(value)
+                    except (ValueError, SyntaxError):
+                        page_tags[name] = value
+                tiff_tags.append(page_tags)
+        return tiff_tags
```

Comment:
considering this is exactly duplicated from the kymo tests, is there a way to move this into `conftest.py`?

I feel like we've run across this kind of thing before but can't remember what we did. I think there's a problem where it's weird syntax to return functions from a fixture, and you want to avoid `from .conftest import ...` @JoepVanlier does this seem familiar to you?

Reply:
I created a fixture in `conftest.py`, and used it in the tests for `kymo` and `scan`. It works without the import.